pcs/lib/cib/resource/stonith.py

```diff
@@ -140,6 +145,39 @@ def validate_stonith_restartless_update(
     return stonith_el, report_list
 
 
+def get_node_key_map_for_mpath(
+    stonith_el: _Element, node_labels: Iterable[str]
+) -> Dict[str, str]:
+    library_error = lambda msg: LibraryError(
+        ReportItem.error(
+            reports.messages.StonithRestartlessUpdateUnableToPerform(msg)
+        )
+    )
+    pcmk_host_map_value = get_value(
+        INSTANCE_ATTRIBUTES_TAG, stonith_el, "pcmk_host_map"
+    )
+    if not pcmk_host_map_value:
+        raise library_error(
+            "mpath reservation keys not found in 'pcmk_host_map' attribute"
+        )
+    node_key_map = {}
+    pattern = re.compile(r"(?P<node>[^=:; \t]+)[=:](?P<key>[^=:; \t]+)[; \t]?")
```

Comment:
Here is pacemaker function which parses pcmk_host_map value: https://github.com/ClusterLabs/pacemaker/blob/main/daemons/fenced/fenced_commands.c#L677-L755
It is not exactly 1:1 copy because i skipped the skipping of escaped characters.